core/src/main/java/org/apache/druid/java/util/common/StringUtils.java

```diff
@@ -153,10 +154,32 @@ public static String toUpperCase(String s)
     return s.toUpperCase(Locale.ENGLISH);
   }
 
+  @Nullable
   public static String urlEncode(String s)
   {
+    if (s == null) {
+      return null;
+    }
+
+    try {
+      // application/x-www-form-urlencoded encodes spaces as "+", but we use this to encode non-form
+      // data as well, so replace "+" with "%20".
```

Comment:
nit: add this comment to javadoc for this method.

Reply:
Added this to method javadoc